examples/step-82/step-82.cc

```diff
@@ -0,0 +1,1500 @@
+          std::pow(p(0) * (1.0 - p(0)) * p(1) * (1.0 - p(1)), 2);
+      }
+    else
+      Assert(false, ExcNotImplemented());
+
+    return return_gradient;
+  }
+
+
+
+  template <int dim>
+  SymmetricTensor<2, dim>
+  ExactSolution<dim>::hessian(const Point<dim> &p,
+                              const unsigned int /*component*/) const
+  {
+    SymmetricTensor<2, dim> return_hessian;
+
+    if (dim == 2)
+      {
+        return_hessian[0][0] = (2.0 - 12.0 * p(0) + 12.0 * p(0) * p(0)) *
+                               std::pow(p(1) * (1.0 - p(1)), 2);
+        return_hessian[0][1] =
+          (2.0 * p(0) - 6.0 * std::pow(p(0), 2) + 4.0 * std::pow(p(0), 3)) *
+          (2.0 * p(1) - 6.0 * std::pow(p(1), 2) + 4.0 * std::pow(p(1), 3));
+        return_hessian[1][1] = (2.0 - 12.0 * p(1) + 12.0 * p(1) * p(1)) *
+                               std::pow(p(0) * (1.0 - p(0)), 2);
+      }
+    else if (dim == 3)
+      {
+        return_hessian[0][0] =
+          (2.0 - 12.0 * p(0) + 12.0 * p(0) * p(0)) *
+          std::pow(p(1) * (1.0 - p(1)) * p(2) * (1.0 - p(2)), 2);
+        return_hessian[0][1] =
+          (2.0 * p(0) - 6.0 * std::pow(p(0), 2) + 4.0 * std::pow(p(0), 3)) *
+          (2.0 * p(1) - 6.0 * std::pow(p(1), 2) + 4.0 * std::pow(p(1), 3)) *
+          std::pow(p(2) * (1.0 - p(2)), 2);
+        return_hessian[0][2] =
+          (2.0 * p(0) - 6.0 * std::pow(p(0), 2) + 4.0 * std::pow(p(0), 3)) *
+          (2.0 * p(2) - 6.0 * std::pow(p(2), 2) + 4.0 * std::pow(p(2), 3)) *
+          std::pow(p(1) * (1.0 - p(1)), 2);
+        return_hessian[1][1] =
+          (2.0 - 12.0 * p(1) + 12.0 * p(1) * p(1)) *
+          std::pow(p(0) * (1.0 - p(0)) * p(2) * (1.0 - p(2)), 2);
+        return_hessian[1][2] =
+          (2.0 * p(1) - 6.0 * std::pow(p(1), 2) + 4.0 * std::pow(p(1), 3)) *
+          (2.0 * p(2) - 6.0 * std::pow(p(2), 2) + 4.0 * std::pow(p(2), 3)) *
+          std::pow(p(0) * (1.0 - p(0)), 2);
+        return_hessian[2][2] =
+          (2.0 - 12.0 * p(2) + 12.0 * p(2) * p(2)) *
+          std::pow(p(0) * (1.0 - p(0)) * p(1) * (1.0 - p(1)), 2);
+      }
+    else
+      Assert(false, ExcNotImplemented());
+
+    return return_hessian;
+  }
+
+
+
+  // @sect3{Implementation of the <code>BiLaplacianLDGLift</code> class}
+
+  // @sect4{BiLaplacianLDGLift::BiLaplacianLDGLift}
+
+  // In the constructor, we set the polynomial degree of the two finite element
+  // spaces, we associate the corresponding DoF handlers to the triangulation,
+  // and we set the two penalty coefficients.
+  template <int dim>
+  BiLaplacianLDGLift<dim>::BiLaplacianLDGLift(const unsigned int fe_degree,
+                                              const double penalty_jump_grad,
+                                              const double penalty_jump_val)
+    : fe(fe_degree)
+    , dof_handler(triangulation)
+    , fe_lift(FE_DGQ<dim>(fe_degree), dim * dim)
+    , dof_handler_lift(triangulation)
```

Comment:
If I understand it correctly, you technically don't need `dof_handler_lift`, but only use it initialize `FEValues` to be used locally. I am not sure but you could hand in to `FEValues::reinit()` a `CellAccessor` instead of a `DoFCellAccessor` and then it should work!? Maybe I missing something? Setting up a system that is 9 times bigger than the actual problem seems to be a pity.

Reply:
And just to be explicit: when @peterrum says `CellAccessor`, what he means is `Triangulation<dim>::cell_iterator` (as opposed to `DoFHandler<dim>::cell_iterator`).

Reply:
I am not sure I understood your comment correctly. We always have that `cell` and `cell_lift` refer to the same cell. However, the local problems we need to solve for the lifting terms (see `$M_cB_c=G_c$` in the introduction) indeed involve tensor-valued functions of size `dim * dim` (4 or 9). In other words, all the lifting terms belong to the FE space `$\mathbb{V}_h^{d\times d}$`.

For simplicity, in this program `$\mathbb{V}_h$` is the same FE space as the one we use for the solution (in particular we use the same FE degree), and we thus have `dim*dim` copies of the FE space we use for the solution. So I think that we could only use `fe` and `dof_handler` (i.e., remove `fe_lift` and `dof_handler_lift`), but then it would basically correspond to do by hand what is done by `FESystem`, which would complicate the description of the computation of `H_h` as well as the implementation...

Please let me know if I misunderstood something or if I am missing something.

Reply:
I have opened a small PR that shows you how to remove the second DoFHandler: https://github.com/dianeguignard/dealii/pull/1